x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/pipeline_select.test.tsx

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { setMockActions, setMockValues } from '../../../../../__mocks__/kea_logic';
+
+import React from 'react';
+
+import { mount, shallow } from 'enzyme';
+
+import { EuiSelectable } from '@elastic/eui';
+
+import { PipelineSelect } from './pipeline_select';
+
+const DEFAULT_VALUES = {
+  addInferencePipelineModal: {
+    configuration: {
+      pipelineName: '',
+    },
+  },
+  existingInferencePipelines: [],
+};
+
+const MOCK_ACTIONS = {
+  selectExistingPipeline: jest.fn(),
+};
+
+describe('PipelineSelect', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    setMockValues({});
+    setMockActions(MOCK_ACTIONS);
+  });
+  it('renders pipeline select with no options', () => {
+    setMockValues(DEFAULT_VALUES);
+
+    const wrapper = shallow(<PipelineSelect />);
+    expect(wrapper.find(EuiSelectable)).toHaveLength(1);
+    const selectable = wrapper.find(EuiSelectable);
+    expect(selectable.prop('options')).toEqual([]);
+  });
+  it('limits pipeline select height to option count', () => {
+    setMockValues({
+      ...DEFAULT_VALUES,
+      existingInferencePipelines: [
+        {
+          modelId: 'model_1',
+          modelType: 'model_1_type',
+          pipelineName: 'pipeline_1',
+          sourceFields: [],
+        },
+        {
+          modelId: 'model_2',
+          modelType: 'model_2_type',
+          pipelineName: 'pipeline_2',
+          sourceFields: [],
+        },
+        {
+          modelId: 'model_3',
+          modelType: 'model_3_type',
+          pipelineName: 'pipeline_3',
+          sourceFields: [],
+        },
+      ],
+    });
+
+    const wrapper = mount(<PipelineSelect />);
+    expect(wrapper.find(EuiSelectable)).toHaveLength(1);
+    const selectable = wrapper.find(EuiSelectable);
+    expect(selectable.prop('height')).toEqual(16 * 6 * 3); // TODO: Use euiTheme.base to calculate height
```

Comment:
Can we please either fix the TODOs or remove them before merging?

Reply:
Good catch, forgot about that TODO. I can't call `const { euiTheme } = useEuiTheme()` in the test because I get this error:

```
Invalid hook call. Hooks can only be called inside of the body of a function component.
```

What are the best workarounds for this? I'd prefer not to hard-code the EUI theme base height.

Reply:
It looks like an issue with the scope of a React `useXxxx()` hook; IIRC I ran into a similar issue when I tried to use `useState()` outside the component that it was used in. Maybe @TattdCodeMonkey knows off the top of his head?

Reply:
I don't believe the euiTheme.base changes too often. It's just a part of the theme to make it easier to change. Given that I would just set a const for the sizes in this file `const smallRowHeight = 16 * 6 * 3;` and use the const here.

Another option would be to import size from `import { euiThemeVars } from '@kbn/ui-theme';` then parse it instead of using `useEuiTheme`. I'm not sure what other trade-offs there are with `@kbn/ui-theme` since I see `useEuiTheme` is used much more.